Add dispersion metric quantiles

rtimpe opened the pull request:
Add `qunatile_coverage`, to enable additional coverage metrics beyond the average coverage.  This gives us a more granular view of the coverage, which will be important for certification.

For example, we may want to look at the 20th percentile of the grid coverage to make sure that it is still sufficiently high.  In the case of the dummy squares system with `T=100` samples, this comes out to .038, meaning that 20% of grid points have no sample point within ~96% of the size of a grid square.  Note that this metric is fairly sensitive to the granularity of the grid (i.e. the `num_steps` parameter).  For comparison, the average coverage in this scenario is .37

jeangud reviewed (approved):
LGTM thank you @rtimpe ! One ask: can you add more context to the PR description:
- Title: Add dispersion metric quantiles
- Why do we want this change?
- Can you add an example result with analysis/plot? What does the result format look like? (array?)

rtimpe commented:
> LGTM thank you @rtimpe ! One ask: can you add more context to the PR description:
> 
>     * Title: Add dispersion metric quantiles
> 
>     * Why do we want this change?
> 
>     * Can you add an example result with analysis/plot? What does the result format look like? (array?)

Updated.  The output is just a single number, so I added an example with a bit of explanation
